lib/std/crypto/utils.zig

```diff
@@ -77,6 +78,66 @@ pub fn timingSafeCompare(comptime T: type, a: []const T, b: []const T, endian: E
     return Order.lt;
 }
 
+/// Add two integers serialized as arrays of the same size, in constant time.
+/// The result is stored into `result`, and `true` is returned if an overflow occurred.
+pub fn timingSafeAdd(comptime T: type, a: []const T, b: []const T, result: []T, endian: Endian) bool {
+    const len = a.len;
+    debug.assert(len == b.len and len == result.len);
+    const bits = switch (@typeInfo(T)) {
+        .Int => |cinfo| if (cinfo.signedness != .unsigned) @compileError("Elements to be added must be unsigned") else cinfo.bits,
+        else => @compileError("Elements to be added must be integers"),
+    };
+    const Cext = std.meta.Int(.unsigned, bits + 1);
```

Comment:
The `+1` trick is incredibly wasteful as it forces the sum to work on 33/65 bits. Use `@addWithOverflow` as it's done in the bigint implementation instead.

Reply:
Is @addWithOverflow guaranteed to be constant-time?

The intent of using a bigger integer is to minimize the chances of having conditional branches.

Reply:
> Is @addWithOverflow guaranteed to be constant-time?

No, but neither is the"bigint" arithmetic that LLVM legalizes this operation to. If the target CPU has a carry flag and a way of transferring it to a GPR, a pretty safe assumption to make, `@addWithOverflow` is cheap and constant-time.

Reply:
The code has been changed to use `@addWithOverflow` and `@subWithOverflow`.

Thanks!